src/Summary.jsx

```diff
@@ -64,6 +81,29 @@ class Summary extends React.Component {
     }
   }
 
+  setColor = (name) => {
+    const { colorPerBiome, colors, featuresCounter } = this.state;
+    this.setState({
+      colorPerBiome: [
+        ...colorPerBiome,
+        {
+          name,
+          color: colors[featuresCounter],
+        },
+      ],
+      featuresCounter: featuresCounter + 1,
+    });
+    return colors[featuresCounter];
```

Comment:
featuresCounter can have the wrong value beacuse the return won't wait until the state has been updated

Reply:
featuresCounter could be a local var in componentDidMount just to assign different colors to the map layer and to keep track of them in colorPerBiome as you are already doing.